### PR TITLE
Fixing recovery mechanism to support multiple journals.

### PR DESCRIPTION
**Pull request by pbarron**
The recovery mechanism for the journal does not work for multiple defined journals. 

The current implementation assumes there is only one journal when recovering. As a result when you define an additional journal the persistent actor fails to recover as it uses the default cassandra-query-journal configuration which points (cassandra-query-journal.write-plugin) to the default journal (cassandra-journal). 

This pull request fixes the problem by making the cassandra-query-journal configurable.

### CODE REVIEW

File: src/main/resources/reference.conf [hunk 1]
```diff
@@ -190,6 +190,9 @@ cassandra-journal {
   # Dispatcher for the plugin actor.
   plugin-dispatcher = "cassandra-journal.default-dispatcher"
 
+  # The query journal to use when recoverying
+  recovery-cassandra-query-journal = "cassandra-query-journal"
+
```
Comment (patriknw) [inline src/main/resources/reference.conf]: This is a great catch. It must be one-to-one between the write and the query plugins. It would be more elegant if we could enforce that in the configuration. I'm thinking if we can do that the query journal is always a sub-section of the write journal config.

```
cassandra-journal {
  query-journal {
  }
}

other-cassandra-journal {
  query-journal {
  }
}
```

Then we would not need these cross references.

To implement that we need: https://github.com/akka/akka/issues/19822

Your pull request is a good first step (to make it work). What do you think about naming the property `query-plugin` to align it with the existing `write-plugin`: https://github.com/pbarron/akka-persistence-cassandra/blob/master/src/main/resources/reference.conf#L432 ?

File: src/main/scala/akka/persistence/cassandra/journal/CassandraRecovery.scala [hunk 2]
```diff
@@ -22,10 +22,10 @@ trait CassandraRecovery extends ActorLogging {
   private[this] val extendedActorSystem = context.system.asInstanceOf[ExtendedActorSystem]
   private implicit val materializer = ActorMaterializer()
 
-  private[this] val queries: CassandraReadJournal =
+  private[this] lazy val queries: CassandraReadJournal =
```
Comment (patriknw) [inline src/main/scala/akka/persistence/cassandra/journal/CassandraRecovery.scala]: why was this needed?

File: src/main/resources/reference.conf (hunk 1 above)
Comment (pbarron) [inline src/main/resources/reference.conf]: ya, certainly would be better to enforce the relationship between the journal and query plugin via the configuration but assume we need to wait for 19822 to be completed first. 

In the interim I have no problem in renaming the config to query-plugin. Will do that now.

File: src/main/scala/akka/persistence/cassandra/journal/CassandraRecovery.scala (hunk 2 above)
Comment (pbarron) [inline src/main/scala/akka/persistence/cassandra/journal/CassandraRecovery.scala]: This is required due to the initialisation order between the CassandraRecovery trait and CassandraJournal which is a self type of CassandraRecovery trait. 

If the queries attribute was defined as a val and not a lazy val it would be initialised before config (accessed on line 28) is initialised in CassandraJournal. The result would be a NPE. 

To get around the initialisation order between CassandraJournal and CassandraRecovery the queries attributes needs to be defined as either as a def or a lazy val so that the queries attribute can initialise after the config is initialise in CassandraJournal. 

The downside of this approach is that you won't know of failures in creating the CassandraReadJournal until it is first used. There is probably better approaches to tackling this but would require some refactoring.

**Comment by patriknw** [comment]
Refs #77
